src/postprocessors/SideCoupledIntegralVariablePostprocessor.C

```diff
@@ -0,0 +1,39 @@
+#include "SideCoupledIntegralVariablePostprocessor.h"
+
+registerMooseObject("MoltresApp", SideCoupledIntegralVariablePostprocessor);
+
+template <>
+InputParameters
+validParams<SideCoupledIntegralVariablePostprocessor>()
+{
+  InputParameters params = validParams<SideIntegralVariablePostprocessor>();
+  params.addRequiredCoupledVar("weight",
+                               "The name of the variable that this postprocessor is integrated with");
+  params.addRequiredParam<PostprocessorName>("divisor",
+                                	     "The postprocessor value of the divisor to normalize by");
+  return params;
+}
+
+SideCoupledIntegralVariablePostprocessor::SideCoupledIntegralVariablePostprocessor(
+    const InputParameters & parameters)
+  : SideIntegralVariablePostprocessor(parameters),
+    _weight(coupledValue("weight")),
+    _divisor(getPostprocessorValue("divisor"))
+{
+  addMooseVariableDependency(mooseVariable());
+}
+
+Real
+SideCoupledIntegralVariablePostprocessor::computeQpIntegral()
+{
+  return _u[_qp] * _weight[_qp] / _divisor;
```

Comment:
Is there a way to make sure that _divisor is never 0?

Reply:
Yea I'll include a MooseError call there to check whether it's 0

Reply:
I've changed reworked this postprocessor to make it easier to understand and use. It now works in conjunction with DivisionPostprocessor to provide flow-averaged quantities or other uses.